### PR TITLE
Add support for global and tracks-only files

### PR DESCRIPTION
**Pull request by SaschaMann**

### CODE REVIEW

**Comment by SaschaMann** [comment]
Example build:

- https://github.com/exercism/org-wide-files/actions/runs/902688455
- Track PR: https://github.com/exercism/julia/pull/404
- Non-Track PR: https://github.com/exercism/Exercism.jl/pull/25

File: .github/workflows/sync.yml [hunk 1]
```diff
@@ -111,6 +111,9 @@ jobs:
     # lets us use actions like actions/checkout.
     # It also gives us a pretty job overview that makes it easy to spot issues with
     # particular tracks.
+    # This approach will stop working once there are >256 repos in the Exercism org.
+    # That's still far away, but once that happens, we can split up the workflow in several workflows,
+    # e.g. one for tracks, one for other repos.
```
Comment (SaschaMann) [inline .github/workflows/sync.yml:116]: I just noticed we're actually above that already. Oops.

Comment (ErikSchierboom) [reply .github/workflows/sync.yml:116]: So many repos!

Comment (ErikSchierboom) [reply .github/workflows/sync.yml:116]: What would be the best approach?

Comment (SaschaMann) [reply .github/workflows/sync.yml:116]: I think we can duplicate the workflow, and have one that exclusively spawns jobs for track repos, and the other that spawns jobs for non-track repos (we should be able to determine this via the Search API). Then keep the rest of the workflow identical for simplicity.

Comment (ErikSchierboom) [reply .github/workflows/sync.yml:116]: 👍 Sounds good

**Comment by SaschaMann** [comment]
For reviewing, I'd recommend you look at the diffs between `sync-rest.yml` and `sync-tooling.yml`/`sync-tracks.yml`

File: .github/workflows/sync-rest.yml [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+
+on:
+  workflow_dispatch:
+    inputs:
+      tracks:
+        description: 'List of tracks to rerun the syncing workflow for, formatted as JSON array, e.g. ["exercism/julia"]'
+        required: true
+        default: ''
+  repository_dispatch:
+    types: [track_changes]
```
Comment (ErikSchierboom) [inline .github/workflows/sync-rest.yml:11]: Do we want to have separate dispatch events now?

File: .github/workflows/sync-rest.yml [hunk 3]
```diff
@@ -0,0 +1,272 @@
+name: Sync Rest
+
+on:
+  workflow_dispatch:
+    inputs:
+      tracks:
+        description: 'List of tracks to rerun the syncing workflow for, formatted as JSON array, e.g. ["exercism/julia"]'
```
Comment (ErikSchierboom) [inline .github/workflows/sync-rest.yml:7]: This applies to the rest of this file: there are quite some mentions of tracks, but this workflow is aimed _not_ to deal with tracks, right?

File: .github/workflows/sync-tooling.yml [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+  GIT_USERNAME: "Exercism Bot"
+  GIT_EMAIL: "66069679+exercism-bot@users.noreply.github.com"
+
+jobs:
+  fetch-tracks:
+    name: Fetch & fork target repos
+    runs-on: ubuntu-20.04
+
+    outputs:
+      tracks: ${{ steps.fetch-tracks.outputs.result }}
```
Comment (ErikSchierboom) [inline .github/workflows/sync-tooling.yml:28]: This still refers to tracks, but this should probably be tooling. Same goes for other mentions in this worklow.

**Review by ErikSchierboom — APPROVED** [review]

File: .github/workflows/sync-rest.yml (hunk 3 above)
Comment (SaschaMann) [reply .github/workflows/sync-rest.yml:7]: I've not renamed most mentions of tracks where they're used as a variable. That should make it easier to apply changes to all workflows and also makes it easier to reason about them. Ideally these should just be called repos but that'd have blown up the diff even more.

File: .github/workflows/sync-rest.yml (hunk 2 above)
Comment (SaschaMann) [reply .github/workflows/sync-rest.yml:11]: It's not necessary, `sync-rest` works for all repos, including tracks, if triggered via dispatch events.

File: .github/workflows/sync-tracks.yml [hunk 5]
```diff
@@ -110,6 +108,9 @@ jobs:
     # lets us use actions like actions/checkout.
     # It also gives us a pretty job overview that makes it easy to spot issues with
     # particular tracks.
+    # This approach will stop working once there are >256 repos in the Exercism org.
+    # That's still far away, but once that happens, we can split up the workflow in several workflows,
+    # e.g. one for tracks, one for other repos.
```
Comment (SaschaMann) [inline .github/workflows/sync-tracks.yml:113]:
```suggestion
```

File: .github/workflows/sync-tooling.yml [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+
+    # Launch one job per track.
+    # This is probably less efficient than running everything in one job
+    # and manually cloning and checking out the repos. However, launching a job
+    # lets us use actions like actions/checkout.
+    # It also gives us a pretty job overview that makes it easy to spot issues with
+    # particular tracks.
+    # This approach will stop working once there are >256 repos in the Exercism org.
+    # That's still far away, but once that happens, we can split up the workflow in several workflows,
+    # e.g. one for tracks, one for other repos.
```
Comment (SaschaMann) [inline .github/workflows/sync-tooling.yml:113]:
```suggestion
```

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**